### DISCUSSION TITLE
API endpoint to check the status of the GoCD pipeline

### DISCUSSION THREAD

**Discussion by mbhatia-mdsol** [gocd/gocd#12547] [Q&A]
I am scheduling a pipeline and I want to fetch status of that run i.e. if it was successful, failed or cancelled. I am able to fetch the pipeline counter(or instance number). But I am not able to get the status. The history object is giving status of each stage and job, but I need the final status of the pipeline. Is there a way to achieve this ?

**Comment by chadlwilson** [comment]
In GoCD pipelines don’t have status of their own - I believe that status has to be inferred from the stages of the pipeline depending on the design of your pipelines/stages etc.

I believe it’s a deliberate decision because stages can have manual gates that complete the stages continuing, can have multiple stages running at once, and technically stages trigger stages in other pipelines (not pipelines having dependency on other pipelines).

I believe you’ll probably need to infer the status based on logic calculated over the individual stage status (or the last stage in the pipeline if that suits your pipeline design), from the pipeline instance or pipeline history apis?
